spring-ai-core/src/main/java/org/springframework/ai/chat/metadata/DefaultUsage.java

```diff
@@ -96,18 +142,18 @@ public boolean equals(Object o) {
 		}
 		DefaultUsage that = (DefaultUsage) o;
 		return Objects.equals(this.promptTokens, that.promptTokens)
-				&& Objects.equals(this.generationTokens, that.generationTokens)
+				&& Objects.equals(this.completionTokens, that.completionTokens)
```

Comment:
equals should also include the `nativeUsage` field in the calculate, as well as for hascode/tostring